ScienceJournal/Sensors/MagnetometerSensor.swift

```diff
@@ -61,15 +62,45 @@ class MagnetometerSensor: MotionSensor {
   }
 
   override func callListenerBlocksWithData(atMilliseconds milliseconds: Int64) {
-    guard let magneticField =
-        MagnetometerSensor.motionManager.deviceMotion?.magneticField.field else { return }
+    guard let magneticField = MagnetometerSensor.motionManager.deviceMotion?.magneticField.field,
+        isValid(magneticFieldReading: magneticField)
+    else { return }
 
     // The magnetic strength is the square root of the sum of the squares of the values in X, Y
     // and Z.
-    let magneticStrength =
+    var magneticStrength =
         sqrt(pow(magneticField.x, 2) + pow(magneticField.y, 2) + pow(magneticField.z, 2))
+
+    // Typical white dwarf stars have a magnetic field of around 100 Tesla (100,000,000 µT), so
+    // let's cap our values there.
+    // Note: Not a good idea to get close enough to measure a white dwarf star.
+    if magneticStrength > whiteDwarfMagneticFieldStrength {
+      magneticStrength = whiteDwarfMagneticFieldStrength
+    }
+
     let dataPoint = DataPoint(x: milliseconds, y: magneticStrength)
     callListenerBlocksWithDataPoint(dataPoint)
   }
 
+  /// Sometimes we can get junk readings. The easiest way to check is to compare all axes
+  /// to make sure their individual readings aren't extremely far apart.
+  private func isValid(magneticFieldReading field: CMMagneticField) -> Bool {
```

Comment:
Sorry for being late with this comment, but this could probably make a nice extension on CMMagneticField `var isValid: Bool`

Reply:
Good idea, will make the change in a separate PR 👍